libs/providers/openstack.py

```
@@ -240,6 +240,15 @@ def clone_vm(
             raise ValueError(f"Could not find flavor for source VM '{source_vm_name}'.")
         flavor_id: str = flavor_obj.id
 
+        # Get source VM's boot volume size (flavor.disk is 0 for volume-backed instances)
+        source_volumes = list(self.api.block_storage.volumes(details=True, attach_to=source_vm.id))
+        boot_volume = next(
+            (vol for vol in source_volumes if any(att.get("boot_index") in (0, "0") for att in vol.attachments)),
+            source_volumes[0] if source_volumes else None,
+        )
+        if not boot_volume:
+            raise ValueError(f"Could not determine boot volume size for '{source_vm_name}'.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)